cardano-cli/src/Cardano/CLI/Types/Common.hs

```diff
@@ -663,3 +666,12 @@ data PotentiallyCheckedAnchor anchorType anchor
   -- ^ Whether to check the hash or not (CheckHash for checking or TrustHash for not checking)
   }
   deriving (Eq, Show)
+
+-- | Type used for serialization when printing the hash of a transaction
+-- after having submitted it.
+newtype TxIdSubmission = TxIdSubmission {txhash :: TxId}
```

Comment:
I still don't like this name 😄

Reply:
The type is unnecessary.

Reply:
If we were using this value in multiple places and could mix it up with other `TxId`s, I would agree with the `newtype` definition. However, it's used in one place, so it's overkill.

Reply:
This newtype is so that we output this JSON on stdout:

```json
{"txhash": "d9290bc8f8f412660109e3c9740d89c19f2ba6697508fd9d4b4b54f32a2ba02e"}
```

If we were to use `TxID`'s JSON format directly, we would have:

```json
"d9290bc8f8f412660109e3c9740d89c19f2ba6697508fd9d4b4b54f32a2ba02e"
```

which we ruled out during our team meeting a few weeks ago (because "it wasn't JSON"). So in the end, I'm not sure what you guys want!?

Reply:
[My gripe is with the variable name.](https://github.com/IntersectMBO/cardano-cli/pull/925#discussion_r1832974843) I understand it currently as "a submission of transaction id" which is vague in my opinion. A better name would be "a result of a transaction submission", for example: `TxSubmissionResult`.

This is non-critical though. I'm leaving an approval to @Jimbo4350 since he blocked the PR originally.

Reply:
Decided during today's team meeting: let's go for the `{"txhash": "d9290bc8f8f412660109e3c9740d89c19f2ba6697508fd9d4b4b54f32a2ba02e"}` and separate JSON from English by using `stdout` and `stderr`, as @carbolymer suggested [here](https://github.com/IntersectMBO/cardano-cli/pull/925#discussion_r1834516926).